src/src/common/middlewares/authmiddleware.ts

```diff
@@ -9,6 +10,10 @@ export class AuthMiddleware implements NestMiddleware {
 		if (req.headers.authorization) {
 			const user = await this.userService.ipUserInfo(req);
 			req.mw_userid = user?.data?.id;
+			if (user) {
+				const decoded: any = jwt_decode(req.headers.authorization);
+				req.mw_role = decoded.resource_access.hasura.roles.filter(role => role !== 'user')[0];
```

Comment:
User may have more than one roles. We should set this as:`req.mw_roles`

Reply:
Yes. Changed to `req.mw_roles`